core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -243,6 +249,9 @@ def __init__(
 
         # self._exceptions_list = []  # TODO: Keep exceptions list for debugging during benchmarking.
 
+        self.callbacks: List[AbstractCallback] = []
+        self._callback_early_stop = False
```

Comment:
I'm not very familiar with the rest of the `AbstractTrainer` code, so this might be a bad suggestion, but would it be feasible to communicate the early stopping / interruption without altering the state of the trainer? I can imagine that in some scenarios such as distributed training it will be really tricky to reason about the state of this variable.

Reply:
Tough to say, but I think I would prefer to edit the trainer state, as the intention is to not pass the trainer object to worker threads, and therefore it remains a singular source of truth. We can change it later if we find this has limitations.